src/lib.zig

```diff
@@ -3015,6 +3027,251 @@ pub const Lua = struct {
     pub fn openBit32(lua: *Lua) void {
         _ = c.luaopen_bit32(lua.state);
     }
+
+    /// Pushes any valid zig value onto the stack,
+    /// Works with ints, floats, booleans, structs,
+    /// optionals, and strings
+    pub fn pushAny(lua: *Lua, value: anytype) void {
+        switch (@typeInfo(@TypeOf(value))) {
+            .Int, .ComptimeInt => {
+                //const casted: Integer = @intCast(value);
+                lua.pushInteger(@intCast(value));
+            },
+            .Float, .ComptimeFloat => {
+                //const casted: Number = @floatCast(value);
+                lua.pushNumber(@floatCast(value));
+            },
+            .Pointer => |info| {
+                switch (info.size) {
+                    .One => {
+                        if (@typeInfo(info.child) == .Array) {
+                            if (@typeInfo(info.child).Array.child != u8) {
+                                @compileError("only u8 arrays can be pushed");
+                            }
+                            _ = lua.pushString(&(value.*));
+                        } else {
+                            if (info.is_const) {
+                                @compileLog(value);
+                                @compileError("Pointer must not be const");
+                            }
+                            lua.pushLightUserdata(@ptrCast(value));
+                        }
+                    },
+                    .C, .Many, .Slice => {
+                        if (info.child != u8) {
+                            @compileError("Only u8 slices (strings) are valid slice types");
+                        }
+                        if (info.sentinel) |sentinel| {
+                            const casted: *info.child = @ptrCast(@constCast(sentinel));
+                            if (casted.* != 0) {
+                                @compileError("Sentinel of slice must be a null terminator");
+                            }
+                            _ = lua.pushString(value);
+                        } else {
+                            var null_terminated = lua.allocator().dupeZ(u8, value) catch unreachable;
```

Comment:
I don't love the `catch unreachable` here. Not sure if this should return a Zig error, or if it should raise a Lua error with `lua.raiseErrorStr`

Reply:
Yes that might have been a temporary solution. I'll resolve this

Reply:
The more I think about it, the more I wonder if this is even something we need to support right now. How often would someone be passing a many-itemed pointer with no sentinel? I think we can just not support that for now. If someone needs it they can do the allocation themselves, or we can add the feature at a later time